### PR TITLE
Use backendconfig v1 for e2e tests

### PR DESCRIPTION
**Pull request by skmatti**
- Existing e2e tests are migrated to use backendconfig v1
- one e2e test for backendconfig v1beta1 API is added

### CODE REVIEW

File: cmd/e2e-test/affinity_beta_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	"k8s.io/ingress-gce/pkg/e2e"
+	"k8s.io/ingress-gce/pkg/fuzz"
+	"k8s.io/ingress-gce/pkg/fuzz/features"
+	testutils "k8s.io/ingress-gce/pkg/test"
+)
+
+func TestAffinityBeta(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range []struct {
```
Comment (freehan) [inline cmd/e2e-test/affinity_beta_test.go]: I am wondering if it is worth merging test cases into one transition test together. Meaning that each test case becomes a state/expectation. The test would go thru each test case, trigger a transition and verify if the final state is expected.

Then there is no need to create separate ingress for each test case.

Comment (skmatti) [reply cmd/e2e-test/affinity_beta_test.go]: Discussed offline that we should keep beta tests in a separate file for simplicity. 

Consolidated affinity transitions to use a single ingress as suggested.

**Pull request merged**